core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -977,33 +997,34 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
   /**
    * Deletes the given zk path recursively
    * @param path
-   * @return true if path gets deleted successfully, false if root path doesn't exists
+   * @return true if path gets deleted successfully, false if root path doesn't exist
    * @throws KeeperException if there is an error while deleting the znodes
    */
   private[zk] def deleteRecursive(path: String): Boolean = {
     val getChildrenResponse = retryRequestUntilConnected(GetChildrenRequest(path))
-    if (getChildrenResponse.resultCode == Code.OK) {
-      getChildrenResponse.children.foreach(child => deleteRecursive(s"$path/$child"))
-      val deleteResponse = retryRequestUntilConnected(DeleteRequest(path, ZkVersion.NoVersion))
-      if (deleteResponse.resultCode != Code.OK && deleteResponse.resultCode != Code.NONODE) {
-        throw deleteResponse.resultException.get
-      }
-      true
-    } else if (getChildrenResponse.resultCode == Code.NONODE) {
-      false
-    } else
-      throw getChildrenResponse.resultException.get
+    getChildrenResponse.resultCode match {
+      case Code.OK =>
+        getChildrenResponse.children.foreach(child => deleteRecursive(s"$path/$child"))
+        val deleteResponse = retryRequestUntilConnected(DeleteRequest(path, ZkVersion.NoVersion))
+        if (deleteResponse.resultCode != Code.OK && deleteResponse.resultCode != Code.NONODE) {
+          throw deleteResponse.resultException.get
+        }
+        true
+      case Code.NONODE =>
+        false
+      case _ =>
+        throw getChildrenResponse.resultException.get
+    }
   }
 
   private[zk] def pathExists(path: String): Boolean = {
-    val getDataRequest = GetDataRequest(path)
-    val getDataResponse = retryRequestUntilConnected(getDataRequest)
-    if (getDataResponse.resultCode == Code.OK) {
-      true
-    } else if (getDataResponse.resultCode == Code.NONODE) {
-      false
-    } else
-      throw getDataResponse.resultException.get
+    val existsRequest = ExistsRequest(path)
+    val existsResponse = retryRequestUntilConnected(existsRequest)
+    existsResponse.resultCode match {
+      case Code.OK => true
+      case Code.NONODE => false
+      case _ => throw existsResponse.resultException.get
+    }
```

Comment:
Can we move the change to use `ExistsRequest` to a separate PR so that it doesn't get lost in the huge refactoring?